enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,186 @@
+---
+title: windows-machine-config-operator
+authors:
+  - "@ravisantoshgudimetla"
+  - "@aravindhp"
+reviewers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@openshift/openshift-team-cloud"
+  - "@openshift/openshift-team-mco"
+approvers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+creation-date: 2020-05-28
+last-updated: 2020-05-28
+status: implementable
+replaces:
+  - "/enhancements/windows-containers/ansible-dev-preview.md"
+---
+
+# Windows Machine Config Operator
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to allow a cluster administrator to add a
+Windows compute node as a day 2 operation with a prescribed configuration to an
+installer provisioned OpenShift 4.6 cluster and enable scheduling of Windows
+workloads. The targeted platforms for 4.6 are vSphere, Azure and AWS. The
+cluster has to be configured with Hybrid OVN networking.
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy customer
+requirement of being able to run Windows workloads on OpenShift clusters.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Provide workflows for adding and removing Windows instances provisioned using
+  the Machine API to OpenShift clusters
+* Perform all the required steps within the VM for it to be added to the cluster
+  as an OpenShift worker node and removed from it.
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Installing the container runtime in the Windows node
+* Upgrades (This will be addressed in a separate enhancement)
```

Comment:
what are the user story and expectations for a cluster that gets upgraded breaking the version compatibility policies between the API server and the kubelet running in a windows VM?